app/src/main/java/com/example/someapplication/data/MoviesRepository.kt

```diff
@@ -11,51 +17,101 @@ import kotlinx.coroutines.withContext
 class MoviesRepository {
 
     private val api = NetworkModule().provideApiService()
+    private val daoMovieDetails = AppDataBase.instance.getMovieDetailsDbDao()
+    private val daoMoviesList = AppDataBase.instance.getMoviesListDbDao()
 
-    suspend fun loadMovies(): List<MoviePreview>? {
-        var moviePreviewList: List<MoviePreview>? = null
+    suspend fun loadMovies(): List<MoviesListEntity>? {
+        var moviePreviewListDto: List<MoviePreview>? = null
+        var moviesResult: List<MoviesListEntity>? = null
         try {
-            moviePreviewList = withContext(Dispatchers.IO) {
+            moviePreviewListDto = withContext(Dispatchers.IO) {
                 api.loadMovies(KEY, LANG, PAGE).results
             }
+            moviesResult = Mapper.mapMoviesListToDb(moviePreviewListDto)
+            withContext(Dispatchers.IO) {
+                daoMoviesList.deleteCachedMoviesList()
+                daoMoviesList.saveMoviesList(moviesResult)
+            }
         } catch (e: Exception) {
         }
```

Comment:
Not be present empty catch block. Add logging information about exceptions. For example, you can use Timber.